src/apps/eth/eth_sign_msg.c

```diff
@@ -33,9 +41,13 @@ app_eth_sign_error_t app_eth_sign_msg(
     if (request->msg.size > 9999) {
         return APP_ETH_SIGN_ERR_INVALID_INPUT;
     }
+    // assert keypath length so that we can access the "network" piece
+    if (request->keypath_count < 2) {
+        return APP_ETH_SIGN_ERR_INVALID_INPUT;
+    }
     // Only support main net for now. Otherwise a user could be tricked into signing something for
     // main net even if they believe they are signing for testnet.
```

Comment:
comment out of date.

> Otherwise a user could be tricked into signing something for
>     // main net even if they believe they are signing for testnet.

This concern is still valid; the coin name should be verified as well, maybe like when verifying the address in the first place (https://github.com/digitalbitbox/bitbox02-firmware/blob/594dd433b22d7db60b12d4b68c9cc977b1bee407/src/commander/commander_eth.c#L28-L30)

What about moving the `display` param from the commander `_api_pub` to `app_eth_address()`, then it comes for free?

Reply:
The problem with messages, unlike transactions, is we can't really verify the coin, because the coin will always match (on MEW specifically), since in the case of messages, the coin is derived from the keypath since we don't have the chainId like for a tx.
So until MEW also provides the device with the network, we can't verify this

Reply:
@benma meant that we should show the network on the device. So the device will print "Ethereum", "Ropsten" or "Rinkeby" depending on which network we intend to sign for. Then the user can abort in case it is wrong.